lib/ansible/modules/cloud/ovirt/ovirt_event.py

```diff
@@ -0,0 +1,250 @@
+                  http://ovirt.github.io/ovirt-engine-api-model/master/#types/event"
+    returned: "On success."
+    type: dict
+'''
+
+import traceback
+
+try:
+    import ovirtsdk4.types as otypes
+except ImportError:
+    pass
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ovirt import (
+    BaseModule,
+    check_sdk,
+    check_params,
+    create_connection,
+    equal,
+    get_dict_of_struct,
+    ovirt_full_argument_spec,
+)
+
+
+class EventsModule(BaseModule):
+
+    def build_entity(self):
+        return otypes.Event(
+            description=self._module.params['description'],
+            severity=otypes.LogSeverity(self._module.params['severity']),
+            origin=self._module.params['origin'],
+            custom_id=self._module.params['custom_id'],
+            id=self._module.params['id'],
+            cluster=otypes.Cluster(
+                id=self._module.params['cluster']
+            ) if self._module.params['cluster'] is not None else None,
+            data_center=otypes.DataCenter(
+                id=self._module.params['data_center']
+            ) if self._module.params['data_center'] is not None else None,
+            host=otypes.Host(
+                id=self._module.params['host']
+            ) if self._module.params['host'] is not None else None,
+            storage_domain=otypes.StorageDomain(
+                id=self._module.params['storage_domain']
+            ) if self._module.params['storage_domain'] is not None else None,
+            template=otypes.Template(
+                id=self._module.params['template']
+            ) if self._module.params['template'] is not None else None,
+            user=otypes.User(
+                id=self._module.params['user']
+            ) if self._module.params['user'] is not None else None,
+            vm=otypes.Vm(
+                id=self._module.params['vm']
+            ) if self._module.params['vm'] is not None else None,
+        )
+
+
+def main():
+    argument_spec = ovirt_full_argument_spec(
+        state=dict(
+            choices=['present', 'absent'],
+            default='present',
+        ),
+        description=dict(default=None),
+        severity=dict(
+            choices=['error', 'normal', 'warning'],
+            default='normal',
+        ),
+        origin=dict(default=None),
+        custom_id=dict(default=None, type='int'),
+        id=dict(default=None),
+        cluster=dict(default=None),
+        data_center=dict(default=None),
+        host=dict(default=None),
+        storage_domain=dict(default=None),
+        template=dict(default=None),
+        user=dict(default=None),
+        vm=dict(default=None),
+    )
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True
+    )
+
+    check_sdk(module)
+
+    # Wait must be set to false if state == absent
+
+    if module.params['state'] == 'absent' and module.params['wait'] is not False:
```

Comment:
What about overriding it always rather than printing error?

Reply:
My thinking was it would force the user to realize wait needed to be set to false. If you want instead of failing I can just do the following on line 224:

`module.params['wait'] = False`

Reply:
In general your solution is good, but user will always have to specify wait=false, unless we fix it in API. But I am ok with current solution, so let's merge.